oaknational/Oak-Web-Application
# LESQ-1113: All uk school URN's provided to Avo

## Description

Music year: 1970

#### When a resource is downloaded or shared, the school URN will be passed to Avo as a string to include NI school URN's 

#### This PR updates the following:
- Updates Avo branch
- Adds two regex functions, to get all UK `schoolUrns` and get `schoolName` from the school Id string

#### Avo events updated:
- curriculumResourcesDownloadedCurriculumDocument
- curriculumResourcesDownloaded
- lessonResourcesDownloaded
- lessonShared

## Issue(s)

Fixes #LESQ-1113

## How to test

1. Go to:

Curriculum downloads:
https://deploy-preview-2879--oak-web-application.netlify.thenational.academy/teachers/programmes/history-primary-ks1/units/traditional-stories-what-do-they-tell-us-about-the-distant-past-1757/lessons/st-patrick-patron-saint-of-ireland/share?preselected=all
https://deploy-preview-2879--oak-web-application.netlify.thenational.academy/teachers/curriculum/previous-downloads?subject=citizenship&keystage=ks4

Teacher downloads / share:
https://deploy-preview-2879--oak-web-application.netlify.thenational.academy/teachers/programmes/chemistry-secondary-ks4-foundation-edexcel/units/atomic-structure-and-the-periodic-table/lessons/atomic-structure-negligible-electron-mass/downloads?preselected=all
https://deploy-preview-2879--oak-web-application.netlify.thenational.academy/teachers/programmes/citizenship-secondary-ks4-l/units/how-well-does-the-media-hold-those-in-power-to-account-63c4/lessons/what-is-the-role-of-the-media-in-a-democracy-cmw32c/share?preselected=all

2. Open console, and click download or share 

3. You should see all download/share events will have have the school URN being sent as a string, Northern Irish schools will be hyphenated

## Screenshots

How it should now look:

![Screenshot 2024-10-15 at 12 04 40](https://github.com/user-attachments/assets/135d027d-5fdd-4b40-99f3-39e2c17e721e)
![Screenshot 2024-10-15 at 12 08 53](https://github.com/user-attachments/assets/5284e035-9b08-49fb-a68e-74a1ed081e53)

## Checklist
